Include CephFS mount options from Manila as posix mount options.

tomclark0 opened the pull request:
This change extracts the `__mount_options` field from the metadata of Manila shares and includes it in the `mount_options` variable.
This field is expected to be populated only with CephFS shares.

Key functionality:
- Pass Manila provided mount options as `mount_options`
- Support for mounting CephFS when multiple backends or multiple CephFS pools are in-use.

This allows the handling of CephFS shares in Manila, where multiple CephFS pools are used within a single Ceph cluster.

sjpb commented on defaults/main.yml:
Defaults are documented in https://github.com/stackhpc/ansible-role-os-manila-mount?tab=readme-ov-file#role-variables, the behaviour of defaults also coming from the share should be documented.

sjpb commented on defaults/main.yml:
```diff
@@ -11,6 +11,7 @@ os_manila_mount_opts:
   - noatime
   - _netdev # prevents mount blocking early boot before networking available
   - rw
+  - "{{ os_manila_mount_share_info[0].mount_options | default('') }}" # Pass mount opts from manila
```
Is it really the desired behaviour that the mount options are only taken from the *first* share and are then used for all? If so, at least I think that needs to be documented.

tomclark0 commented:
Thanks @sjpb.
I've changed approach and removed the user facing var as a duplicate.
Any manila provided mount options either manually set via `openstack share set --property "__mount_options=" share_id` or added due to different backends are honoured on a per-share basis.

sjpb commented on tasks/mount.yml:
```diff
@@ -46,7 +46,7 @@
     src: "{{ item[1].host }}:{{ item[1].export }}"
     fstype: ceph
-    opts: "name={{ item[1].share_user }},{{ (item[0].mount_opts | default(os_manila_mount_opts)) | join(',') }}"
+    opts: "name={{ item[1].share_user }},{{ item[1].mount_options | default('') }},{{ (item[0].mount_opts | default(os_manila_mount_opts)) | join(',') }}"
```
This will result in e.g.

`opts: "name=$user,,$os_manila_mount_opts" if there are no mount options from the share` - is that empty string ok?

tomclark0 replied:
The empty string doesn't affect the mount command. I don't have an environment to confirm if these mount opts are always defined.

sjpb reviewed (approved):
LGTM
